packages/dds/tree/src/shared-tree-core/editManager.ts

```diff
@@ -705,52 +703,47 @@ export class EditManager<
 			return;
 		}
 
-		// Remote changes, i.e., changes from remote clients are applied in three steps.
-		for (const newCommit of newCommits) {
-			// Step 1 - Recreate the peer remote client's local environment.
-			// Get the revision that the remote change is based on
-			const [, baseRevisionInTrunk] = this.getClosestTrunkCommit(referenceSequenceNumber);
-			// Rebase that peer local branch over the part of the trunk up to the base revision
-			// This will be a no-op if the sending client has not advanced since the last time we received an edit from it
-			const peerLocalBranch = getOrCreate(
-				this.peerLocalBranches,
-				sessionId,
-				() =>
-					new SharedTreeBranch(baseRevisionInTrunk, this.changeFamily, this.mintRevisionTag),
-			);
-			peerLocalBranch.rebaseOnto(this.trunk, baseRevisionInTrunk);
+		/* Remote changes, i.e., changes from remote clients are applied in three steps. */
+
+		// Step 1 - Recreate the peer remote client's local environment.
+		// Get the revision that the remote change is based on
+		const [, baseRevisionInTrunk] = this.getClosestTrunkCommit(referenceSequenceNumber);
+		// Rebase that peer local branch over the part of the trunk up to the base revision
+		// This will be a no-op if the sending client has not advanced since the last time we received an edit from it
+		const peerLocalBranch = getOrCreate(
+			this.peerLocalBranches,
+			sessionId,
+			() => new SharedTreeBranch(baseRevisionInTrunk, this.changeFamily, this.mintRevisionTag),
+		);
+		peerLocalBranch.rebaseOnto(this.trunk, baseRevisionInTrunk);
 
-			// Step 2 - Append the change to the peer branch. Rebase the change to the tip of the trunk.
-			if (peerLocalBranch.getHead() === this.trunk.getHead()) {
-				// If the branch is fully caught up and empty after being rebased, then push to the trunk directly
+		// Step 2 - Append the changes to the peer branch and rebase the changes to the tip of the trunk.
+		if (peerLocalBranch.getHead() === this.trunk.getHead()) {
+			// If the peer local branch is fully caught up and empty (no changes relative to the trunk) after being
+			// rebased, then push to changes the trunk directly and update the peer branch to the trunk's head.
+			for (const newCommit of newCommits) {
 				this.pushCommitToTrunk(nextSequenceId, { ...newCommit, sessionId });
 				peerLocalBranch.setHead(this.trunk.getHead());
```

Comment:
`peerLocalBranch.setHead(this.trunk.getHead());` could be pulled out of the loop and done once after it.